pkg/providers/ironic.go

```diff
@@ -305,6 +310,11 @@ func (p *ironicProvider) selectNode(poolType string) (*nodes.Node, error) {
 
 		for x := 0; x < len(thenodes); x++ {
 			node := thenodes[x]
+
+			if node.ProvisionState != string(nodes.Active) && node.ProvisionState != string(nodes.Available) {
```

Comment:
Which issue was addressing this line exactly? Slow refresh maybe?

Reply:
The case I hit was a that a number of nodes were available to be picked by the provider
the provider pick on that was in a failed state it should have skipped it and picked on that was available

Reply:
Got it, thanks